src/main/java/edu/uw/edm/edmzuulproxy/healthcheck/GwsHealthIndicator.java

```diff
@@ -1,27 +1,22 @@
 package edu.uw.edm.edmzuulproxy.healthcheck;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.actuate.health.Health;
 import org.springframework.boot.actuate.health.HealthIndicator;
 import org.springframework.stereotype.Component;
-import org.springframework.beans.factory.annotation.Autowired;
-
-import java.util.List;
-
-import edu.uw.edm.gws.GroupsWebServiceClient;
-import edu.uw.edm.gws.model.GWSSearchType;
-import edu.uw.edm.gws.model.GroupReference;
 
+import edu.uw.edm.edmzuulproxy.security.GroupsResolver;
 import lombok.extern.slf4j.Slf4j;
 
 @Component
 @Slf4j
 public class GwsHealthIndicator implements HealthIndicator {
 
-    final private GroupsWebServiceClient groupsWebServiceClient;
+    private final GroupsResolver groupsResolver;
 
     @Autowired
-    public GwsHealthIndicator(GroupsWebServiceClient groupsWebServiceClient) {
-        this.groupsWebServiceClient = groupsWebServiceClient;
+    public GwsHealthIndicator(GroupsResolver groupsResolver) {
```

Comment:
Not sure the reason for using GroupResolver, instead of GroupsWebServiceClient.
Using the GroupResolver requires more boilerplate code in GroupResolver files. Using GroupsWebServiceClient directly reduces the amount of boilerplate code and touches fewer files.

Reply:
Caching with the @Cachable is handled at the interface/proxy level, it can't be within the same class.